From 4bd23c637585903378fcd35828cd9d0e01d7d565 Mon Sep 17 00:00:00 2001
From: Michael-CY Lee <michael-cy.lee@mediatek.com>
Date: Mon, 25 Mar 2024 18:59:35 +0800
Subject: [PATCH 57/61] mtk: wifi: mt76: mt7996: not to check 'need_offchan'
 for MLD multicast mgmt.

Multicast mgmt. sent by the MLD AP should be transmitted via all links,
so it is not necessary to check 'need_offchan'.

CR-Id: WCNCR00289305
Signed-off-by: Michael-CY Lee <michael-cy.lee@mediatek.com>
Change-Id: I850d337ea98456b8d81fd767ceea1727333aa6ca
---
 net/mac80211/offchannel.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/net/mac80211/offchannel.c b/net/mac80211/offchannel.c
index ec1d7a1..a571be0 100644
--- a/net/mac80211/offchannel.c
+++ b/net/mac80211/offchannel.c
@@ -874,7 +874,8 @@ int ieee80211_mgmt_tx(struct wiphy *wiphy, struct wireless_dev *wdev,
 	/* Check if the operating channel is the requested channel */
 	if (!params->chan && mlo_sta) {
 		need_offchan = false;
-	} else if (!need_offchan) {
+	} else if (!need_offchan && !(ieee80211_vif_is_mld(&sdata->vif) &&
+		   is_multicast_ether_addr(mgmt->da))) {
 		struct ieee80211_chanctx_conf *chanctx_conf = NULL;
 		int i;
 
-- 
2.39.2

